test/infrastructure/docker/api/v1beta1/conversion.go

```diff
@@ -83,11 +87,18 @@ func (src *DockerMachine) ConvertTo(dstRaw conversion.Hub) error {
 	}
 
 	restored := &infrav1.DockerMachine{}
-	if ok, err := utilconversion.UnmarshalData(dst, restored); err != nil || !ok {
+	ok, err := utilconversion.UnmarshalData(src, restored)
+	if err != nil {
 		return err
 	}
 
-	dst.Status.Initialization = restored.Status.Initialization
+	// Recover intent for bool values converted to *bool.
+	initialization := infrav1.DockerMachineInitializationStatus{}
+	restoredDockerMachineProvisioned := restored.Status.Initialization.Provisioned
```

Comment:
nit: not dockerMachine. Please check all these variables in this file

Reply:
Here is DockerMachine. L-157 is incorrect (correctly it's DevCluster)

Reply:
Ah sorry added the comment to the wrong occurence :)